src/components/common/ColonyActions/CountDownTimer/helpers.ts

```diff
@@ -0,0 +1,188 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+
+import {
+  GetMotionTimeoutPeriodsReturn,
+  useGetMotionTimeoutPeriodsQuery,
+} from '~gql';
+import { RefetchMotionState, useAppContext, useColonyContext } from '~hooks';
+import { MotionState } from '~utils/colonyMotions';
+import { MotionData } from '~types';
+
+const splitTimeLeft = (period: number) => ({
+  days: Math.floor(period / (60 * 60 * 24)),
+  hours: Math.floor((period / (60 * 60)) % 24),
+  minutes: Math.floor((period / 60) % 60),
+  seconds: Math.floor(period % 60),
+});
+
+const getCurrentStatePeriodInMs = (
+  motionState: MotionState,
+  motionTimeoutPeriods: GetMotionTimeoutPeriodsReturn,
+) => {
+  switch (motionState) {
+    case MotionState.Staking:
+    case MotionState.Staked:
+    case MotionState.Objection:
+      return motionTimeoutPeriods.timeLeftToStake;
+    case MotionState.Voting:
+      return motionTimeoutPeriods.timeLeftToVote;
+    case MotionState.Reveal:
+      return motionTimeoutPeriods.timeLeftToReveal;
+    case MotionState.Escalation:
+      return motionTimeoutPeriods.timeLeftToEscalate;
+    default:
+      return '-1';
+  }
+};
+
+const useMotionTimeoutPeriods = (colonyAddress = '', motionId: string) => {
+  const { data, loading, refetch } = useGetMotionTimeoutPeriodsQuery({
+    variables: { input: { colonyAddress, motionId } },
+    notifyOnNetworkStatusChange: true,
+    fetchPolicy: 'network-only',
+  });
+  const motionTimeoutPeriods = data?.getMotionTimeoutPeriods || {
+    timeLeftToEscalate: '-1',
+    timeLeftToVote: '-1',
+    timeLeftToReveal: '-1',
+    timeLeftToStake: '-1',
+  };
+
+  return {
+    motionTimeoutPeriods,
+    loadingMotionTimeoutPeriods: loading,
+    refetchMotionTimeoutPeriods: refetch,
+  };
+};
+
+export const useMotionCountdown = (
+  state: MotionState,
+  motionId: string,
+  refetchMotionState: RefetchMotionState,
+  motionStakes: MotionData['motionStakes'],
```

Comment:
Codegen generates a `MotionStakes` type for us, I think that's generally the way we're typing nested objects (As opposed to accessing as a property of the top level type)